lib/vapor/watch.ex

```diff
@@ -10,20 +10,34 @@ defmodule Vapor.Watch do
   end
 
   def init(state) do
+    state = Map.merge(%{error_count: 0, current_alarm: false}, state)
     schedule_refresh(state.plan)
     {:ok, state}
   end
 
   def handle_info(:refresh, %{plan: plan, store: store, layer: layer}=state) do
     case Provider.load(plan.provider) do
       {:ok, new_config} ->
+        # We don't mind if we repeatedly spam the logs when there's a problem,
+        # but if everything is okay, we ought not attempt to clear an alarm
+        # that might not exist because the attempt with spam the logs.
+        if state.current_alarm do
```

Comment:
I wonder if we want to only clear this if we've seen a number of good results come through. That way we don't cause the alarm to flap. But now that I'm saying that maybe its not useful since the watches are only going to fire intermittently. Its probably something we can incorporate at a later date if we find it necessary. What do you think?